### PR TITLE
Improved and more Coordinators

### PR DESCRIPTION
**Pull request by Alexwijn**
## Release Notes

- **New Features**
  - Enhanced configuration options for MQTT, OpenTherm, and EMS-ESP gateways.
  - Added error monitoring capabilities.
  - Introduced a structured approach for managing heating across multiple areas.
  
- **Improvements**
  - Updated translation files for better clarity in multiple languages.
  - Improved logging and error handling for various components.
  - Enhanced test scenarios with new configuration parameters.

- **Bug Fixes**
  - Resolved issues related to device identification and configuration flow.

- **Chores**
  - Updated dependencies and version numbers across various configuration files.

### CODE REVIEW

**Comment by Alexwijn** [comment]
@coderabbitai review

**Comment by Alexwijn** [comment]
@coderabbitai review

**Comment by Alexwijn** [comment]
@coderabbitai review

**Comment by Alexwijn** [comment]
@coderabbitai review

**Comment by Alexwijn** [comment]
@coderabbitai review

**Comment by Alexwijn** [comment]
@coderabbitai summary

File: custom_components/sat/mqtt/ems.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        await self._publish_command(f'{{"cmd": "dhw/seltemp", "value": {value}}}')
+
+        await super().async_set_control_hot_water_setpoint(value)
+
+    async def async_set_control_thermostat_setpoint(self, value: float) -> None:
+        # Not supported (yet)
+        await super().async_set_control_thermostat_setpoint(value)
+
+    async def async_set_heater_state(self, state: DeviceState) -> None:
+        await self._publish_command(f'{{"cmd: "forceheatingoff", "value": {DATA_OFF if state == DeviceState.ON else DATA_ON}}}')
```
Comment (ahhoj) [inline custom_components/sat/mqtt/ems.py:121]:
```suggestion
        await self._publish_command(f'{{"cmd": "heatingoff", "value": "{DATA_OFF if state == DeviceState.ON else DATA_ON}"}}')
```

As per local try, all change on this line is requierd for this to actually work

Comment (Alexwijn) [reply custom_components/sat/mqtt/ems.py]: Ah thanks!

**Comment by Alexwijn** [comment]
@coderabbitai review

**Comment by Alexwijn** [comment]
@coderabbitai summary

File: custom_components/sat/mqtt/ems.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+DATA_CONTROL_SETPOINT = "selflowtemp"
+DATA_REL_MOD_LEVEL = "curburnpow"
+DATA_BOILER_TEMPERATURE = "curflowtemp"
+DATA_RETURN_TEMPERATURE = "rettemp"
+
+DATA_DHW_ENABLE = "tapwateractive"
+DATA_CENTRAL_HEATING = "heatingactive"
+DATA_BOILER_CAPACITY = "nompower"
+
+DATA_REL_MIN_MOD_LEVEL = "burnminnpower"
```
Comment (ahhoj) [inline custom_components/sat/mqtt/ems.py:25]:
```suggestion
DATA_REL_MIN_MOD_LEVEL = "burnminpower"
```

File: custom_components/sat/mqtt/ems.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    def device_active(self) -> bool:
+        return bool(self.data.get(DATA_CENTRAL_HEATING))
+
+    @property
+    def flame_active(self) -> bool:
+        return bool(self.data.get(DATA_FLAME_ACTIVE))
+
+    @property
+    def hot_water_active(self) -> bool:
+        return bool(self.data.get(DATA_DHW_ENABLE))
```
Comment (ahhoj) [inline custom_components/sat/mqtt/ems.py:60]:
```suggestion
    @property
    def device_active(self) -> bool:
        return self.data.get(DATA_CENTRAL_HEATING) == DATA_ON

    @property
    def flame_active(self) -> bool:
        return self.data.get(DATA_FLAME_ACTIVE) == DATA_ON

    @property
    def hot_water_active(self) -> bool:
        return self.data.get(DATA_DHW_ENABLE) == DATA_ON
```

File: custom_components/sat/mqtt/ems.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        await self._publish_command(f'{{"cmd": "dhw/seltemp", "value": {value}}}')
+
+        await super().async_set_control_hot_water_setpoint(value)
+
+    async def async_set_control_thermostat_setpoint(self, value: float) -> None:
+        # Not supported (yet)
+        await super().async_set_control_thermostat_setpoint(value)
+
+    async def async_set_heater_state(self, state: DeviceState) -> None:
+        await self._publish_command(f'{{"cmd": "heatingoff", "value": "{DATA_OFF if state == DeviceState.ON else DATA_ON}"}}')
```
Comment (ahhoj) [inline custom_components/sat/mqtt/ems.py:121]: when sending this, the already configured seleted flow temp will reset to the maximum heating temperature, hence cannot control the selected flow temp appropriately
I truly believe this should be sent only when the the heating needs to be turned on, or off. And not regularly

Comment (Alexwijn) [reply custom_components/sat/mqtt/ems.py]: Sounds good!

File: custom_components/sat/mqtt/ems.py (hunk 3 above)
Comment (Alexwijn) [reply custom_components/sat/mqtt/ems.py]: Thanks!

File: custom_components/sat/mqtt/ems.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+        await self._publish_command(f'{{"cmd": "dhw/seltemp", "value": {value}}}')
+
+        await super().async_set_control_hot_water_setpoint(value)
+
+    async def async_set_control_thermostat_setpoint(self, value: float) -> None:
+        # Not supported (yet)
+        await super().async_set_control_thermostat_setpoint(value)
+
+    async def async_set_heater_state(self, state: DeviceState) -> None:
+        if (state == DeviceState.ON) != self.device_active:
```
Comment (ahhoj) [inline custom_components/sat/mqtt/ems.py:121]: Looks like there is a problem with this solution. It checks heatingactive property, but it is also in 'off' state, when the forceheating is set to off, but the current flow temp overrun the currently configured setpoint, so the burner is turned off on the boiler.
In other words, forceheatingoff is still set to off, so requesting for heat, but temporarily no actual vurning happens, only the pump runs. And as a result, the earlier problematic interference with this setting still occurs.
This check should explicitly go against the currently forceheatingoff data in my opinion.

File: custom_components/sat/mqtt/ems.py [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+DATA_BOILER_DATA = "boiler_data"
+DATA_FLAME_ACTIVE = "burngas"
+DATA_DHW_SETPOINT = "dhw/seltemp"
+DATA_CONTROL_SETPOINT = "selflowtemp"
+DATA_REL_MOD_LEVEL = "curburnpow"
+DATA_BOILER_TEMPERATURE = "curflowtemp"
+DATA_RETURN_TEMPERATURE = "rettemp"
+
+DATA_DHW_ENABLE = "tapactivated"
+DATA_CENTRAL_HEATING = "heatingactive"
```
Comment (ahhoj) [inline custom_components/sat/mqtt/ems.py:22]:
```suggestion
DATA_CENTRAL_HEATING = "heatingactive"
DATA_DESIRED_HEATING_ACTIVE = "heatingoff"
```

File: custom_components/sat/mqtt/ems.py [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+    def supports_maximum_setpoint_management(self) -> bool:
+        return True
+
+    @property
+    def supports_relative_modulation_management(self) -> bool:
+        return True
+
+    @property
+    def device_active(self) -> bool:
+        return self.data.get(DATA_CENTRAL_HEATING) == DATA_ON
```
Comment (ahhoj) [inline custom_components/sat/mqtt/ems.py:52]:
```suggestion
    @property
    def desired_heating_active(self) -> bool | None:
        return None if self.data.get(DATA_DESIRED_HEATING_ACTIVE) is None else self.data.get(DATA_DESIRED_HEATING_ACTIVE) == DATA_OFF

    @property
    def device_active(self) -> bool:
        return self.data.get(DATA_CENTRAL_HEATING) == DATA_ON
```

File: custom_components/sat/mqtt/ems.py (hunk 5 above)
Comment (ahhoj) [reply custom_components/sat/mqtt/ems.py:121]:
```suggestion
if self.desired_heating_active is not None and (state == DeviceState.ON) != self.desired_heating_active:
```

**Review by ahhoj — COMMENTED** [review]
Added 3 seperate suggestiosn which could handle the heatingoff problem in theory.

**Comment by Alexwijn** [comment]
@coderabbitai full

**Comment by Alexwijn** [comment]
@coderabbitai resolve

**Pull request merged**